src/Internal/Service/PaymentProcessingService.php

```diff
@@ -8,19 +8,46 @@
 namespace WCPay\Internal\Service;
 
 use Exception; // Temporary exception! This service would have its own exception when more business logics are added.
+use WCPay\Vendor\League\Container\Exception\ContainerException;
+use WCPay\Internal\Payment\PaymentContext;
+use WCPay\Internal\Payment\State\InitialState;
+use WCPay\Internal\Payment\State\StateFactory;
+use WCPay\Internal\Payment\Exception\StateTransitionException;
 
 /**
  * Payment Processing Service.
  */
 class PaymentProcessingService {
+	/**
+	 * Factory for states.
+	 *
+	 * @var StateFactory
+	 */
+	private $state_factory;
+
+	/**
+	 * Service constructor.
+	 *
+	 * @param StateFactory $state_factory Factory for payment states.
+	 */
+	public function __construct( StateFactory $state_factory ) {
+		$this->state_factory = $state_factory;
+	}
+
 	/**
 	 * Process payment.
 	 *
 	 * @param  int $order_id Order ID provided by WooCommerce core.
 	 *
 	 * @throws Exception
+	 * @throws StateTransitionException In case a state cannot be initialized.
+	 * @throws ContainerException       When the dependency container cannot instantiate the state.
 	 */
 	public function process_payment( int $order_id ) {
-		throw new Exception( 'Re-engineering payment process is in-progress. Sit tight, and wait more!' );
+		$context = new PaymentContext();
+		$state   = $this->state_factory->create_state( InitialState::class, $context );
+		$state   = $state->process();
+
+		return $state;
```

Comment:
Now, it's simple and straightforward. But what if this goes through a few internal states, how will this can be handled? 

I think this question is somewhat mentioned here https://github.com/Automattic/woocommerce-payments/pull/7318#discussion_r1342721335 but it's still not clear to me how we're going to do that.

Reply:
Actually, the beauty here is that if necessary, we can do that from state to state:

```php
class InitialState {
	public function process() {
		$next_state = $this->create_state( ProcessedState::class );
		return $next_state->complete();
	}
}
```

This allows us to define some discrete states, which the payment can go through, without those being visible to the outside.

Reply:
Gotcha! I understand that now. But I hope that we keep using `$next_state->process()`, which has been described in `AbstractPaymentState`, rather than a new method such as `$next_state->complete()`.

Reply:
We'll discuss that when we get there, but why not? The next state would not be doing the same thing, and considering that most of those would be called internally, it would make sense to name the method as what it does.